### ISSUE TITLE
Turn on/off gradients computation between generator/discriminator

### ISSUE THREAD

**Issue by inspirit** [lucidrains/gigagan-pytorch#43]
Hi! was reviewing some code parts of your implementation (great as usual!) and noticed that you dont alter gradient computation for models when its not required. for example computing losses for discriminator we dont need gradients in generator model and vice versa, i believe it should save some memory?

**Comment by lucidrains** [comment]
@inspirit yea, i think i have it for the generator when training the discriminator https://github.com/lucidrains/gigagan-pytorch/blob/main/gigagan_pytorch/gigagan_pytorch.py#L2187

i'll look into adding the other one later today

**Comment by inspirit** [comment]
yup, when training generator it looks good, i was referring generator training block, sorry for the confusion
